src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -44,11 +67,26 @@ public void observeDangerZone(MovedEvent event, EntityRef player) {
         if (height < thresholdHeight && lastHeight >= thresholdHeight) {
             if (player.hasComponent(HypothermiaComponent.class)) {
                 player.removeComponent(HypothermiaComponent.class);
+                delayManager.cancelPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID);
             }
         }
     }
 
     @ReceiveEvent(components = {HypothermiaComponent.class})
+    public void onHypothermia(OnAddedComponent event, EntityRef player) {
+        delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, initialDelay, healthDecreaseInterval);
+    }
```

Comment:
Repeating what I just wrote on the other PR regarding visible breath here. On a second thought, I'd come to the conclusion that all this might be better off in a separate system for the same reason as explained in the other PR (decoupling).